### PR TITLE
Warn on other declarations that default to integer type

### LINKED ISSUE #225: Missing warning for `*p;`
```c
i;
*p;
<stdin>:2:1: warning: type specifier missing, defaults to int
i;
^
```

### PR DESCRIPTION
**Pull request by repnop**
Fixes #225

### CODE REVIEW

File: src/parse/decl.rs [hunk 1]
```diff
@@ -445,10 +445,13 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
                 if signed.is_none() {
                     // if it's not an id, it's invalid anyway
```
Comment (jyn514) [inline src/parse/decl.rs]:
```suggestion
                    // these are the only tokens that can come before an id in a declarator
                    // if it's not a declarator, it's invalid anyway
```

Comment (repnop) [reply src/parse/decl.rs]: Changed :)

File: src/parse/decl.rs [hunk 2]
```diff
@@ -443,8 +443,10 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
             Some(ctype) => ctype,
             None => {
                 if signed.is_none() {
-                    // if it's not an id, it's invalid anyway
-                    // other parts of the parser will have a better error message
+                    // these are the only tokens that can come before an id in a
+                    // declarator if it's not a declarator, it's invalid anyway
```
Comment (jyn514) [inline src/parse/decl.rs]: You line wrapped it and it changed the meaning :(
```suggestion
                    // these are the only tokens that can come before an id in a declarator
                    // if it's not a declarator, it's invalid anyway
```

Comment (repnop) [reply src/parse/decl.rs]: oh no! hecc

**Pull request merged**